extension/src/Xliff/XLIFFCache.ts

```diff
@@ -0,0 +1,78 @@
+import { Xliff } from "./XLIFFDocument";
+import * as path from "path";
+import { InvalidXmlError } from "../Error";
+import { logger } from "../Logging/LogHelper";
+import * as SettingsLoader from "../Settings/SettingsLoader";
+
+class XliffCache {
+  private cache: Map<string, Xliff>;
+  private enabled: boolean;
+
+  public get size(): number {
+    return this.cache.size;
+  }
+
+  public get isEnabled(): boolean {
+    return this.enabled;
+  }
+
+  constructor() {
+    this.cache = new Map();
+    this.enabled = SettingsLoader.getSettings().enableXliffCache;
+  }
+
+  get(filePath: string): Xliff {
+    const fileName = path.basename(filePath);
+    const xliffDocument = this.isCached(filePath)
+      ? this.cache.get(fileName)
+      : this.read(filePath);
+
+    if (xliffDocument) {
+      if (this.enabled) {
+        this.cache.set(fileName, xliffDocument);
+      }
+      return xliffDocument;
+    }
+    throw new Error(`${fileName} not found.`);
```

Comment:
I think you might need to elaborate on this one. I kind of get what you mean but I can't really see what the difference would be, so there's something I'm missing :sweat_smile:

Reply:
It's just that the error could never be hit :)

Nevermind, I think it's fine now

Reply:
`this.cache.get` actually has the return signature of `Xliff | undefined` so typescript would have opinions either way :)

Reply:
> `this.cache.get` actually has the return signature of `Xliff | undefined` so typescript would have opinions either way :)

Yes, that was the reason I suggested to skip the `isCached` call, since we did the test on the results of `get()` anyways :)

Reply:
Well I kept staring on it because I had the distinct feeling that I could comply with your wish but I couldn't just see how to do it. But suddenly it came to me. See the latest commit :+1: It literally took this long for the penny to drop.